ethers-providers/src/gas_oracle/etherscan.rs

```diff
@@ -60,6 +60,12 @@ impl Etherscan {
 #[async_trait]
 impl GasOracle for Etherscan {
     async fn fetch(&self) -> Result<U256, GasOracleError> {
+        match self.gas_category {
+            GasCategory::Fast => return Err(GasOracleError::GasCategoryNotSupported),
+            GasCategory::Fastest => return Err(GasOracleError::GasCategoryNotSupported),
+            _ => {}
+        };
```

Comment:
I believe this can be rewritten with `if matches!(self.category, GasCategory::Fast || GasCategory::Fast) { return Err(...) }`

https://doc.rust-lang.org/beta/std/macro.matches.html

Reply:
More or less personal codestyle preference. I'd say 2-3 specific matches and `_ => {}` are more readable than macro. But `GasCategory::Fast || GasCategory::Fast` should be used anyway in such cases.